lib/sparrow/pools_warden.ex

```diff
@@ -94,33 +94,43 @@ defmodule Sparrow.PoolsWarden do
       end)
   end
 
-  @spec handle_info(any, :ok) :: {:noreply, :ok}
-  def handle_info(unknown, _state) do
+  @impl GenServer
+  def handle_info({:DOWN, _ref, :process, pid, reason}, state) do
+    {pid, pool_type, pool_name, tags} = List.keyfind(state, pid, 0)
+    :ets.delete_object(@tab_name, {pool_type, {pool_name, tags}})
+
+    _ =
+      Logger.info(fn ->
+        "worker=pools_warden, action=unregistering pool, pool_name=#{inspect(pool_name)}, pid=#{inspect(pid)},
```

Comment:
You recommend not to use pool name but tags instead, but than you log pool name but not tags.
Maybe just log whole ets object?